File: Keynote/Python/Diffusion.py
import numpy as np

NumberOfNeighbors = np.array([2, 2, 2, 2, 2])
Milk = np.array([0.0, 0.0, 1.0, 0.0, 0.0])
print Milk

for i in range(0, 10):
  ShareTheMilk = Milk/NumberOfNeighbors
  Milk[0] = ShareTheMilk[4] + ShareTheMilk[1]
  Milk[4] = ShareTheMilk[3] + ShareTheMilk[0]
  
  Milk[1:4] = ShareTheMilk[0:3] + ShareTheMilk[2:5]
  print Milk

File: Keynote/R/Diffusion.R
NumberOfNeighbors = c(2, 2, 2, 2, 2)
Milk = c(0, 0, 1, 0, 0)
barplot(Milk, ylim = c(0, 1))

for (i in 1:80) {
    ShareTheMilk = Milk/NumberOfNeighbors
    # left boundary, index starts at 1
    Milk[1] = ShareTheMilk[5] + ShareTheMilk[2]
    # right boundary
    Milk[5] = ShareTheMilk[4] + ShareTheMilk[1]
    Milk[2:4] = ShareTheMilk[1:3] + ShareTheMilk[3:5]
    print(Milk)
    barplot(Milk, ylim = c(0, 1))
}
